phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/apps/AppsController.java

```diff
@@ -72,12 +72,29 @@ public void loadApps() {
         });
 
         this.view.setOnSelectScript(
-                scriptDTO -> scriptInterpreter.runScript(scriptDTO.getScript(), e -> Platform.runLater(() -> {
-                    // no exception if installation is cancelled
-                    if (!(e.getCause() instanceof InterruptedException)) {
-                        new ErrorMessage(tr("The script ended unexpectedly"), e, this.view);
-                    }
-                })));
+                scriptDTO -> {
+                    final StringBuilder environmentBuilder = new StringBuilder();
+                    environmentBuilder.append("TYPE_ID=\"");
+                    environmentBuilder.append(scriptDTO.getTypeId());
+                    environmentBuilder.append("\";\n");
+                    environmentBuilder.append("CATEGORY_ID=\"");
+                    environmentBuilder.append(scriptDTO.getCategoryId());
+                    environmentBuilder.append("\";\n");
+                    environmentBuilder.append("APPLICATION_ID=\"");
+                    environmentBuilder.append(scriptDTO.getApplicationId());
+                    environmentBuilder.append("\";\n");
+                    environmentBuilder.append("SCRIPT_ID=\"");
+                    environmentBuilder.append(scriptDTO.getId());
+                    environmentBuilder.append("\";\n");
+                    final String environment = environmentBuilder.toString();
```

Comment:
Why don't we inject the dtos directly?

Reply:
If we do that, you would have direct access to more or less the complete repository. I'm not sure that's a good idea. Also it feels a bit strange to me to store objects in environment variables. If a script needs to access the DTO, it can still use e.g. `getApplication()`.

Reply:
I think we should inject a context object, that provides a getter method for all important context information, i.e. the typeId, scriptId, applicationId. These getter functions could then do the lookup. The main benefit of this is, that we can hide the lookup logic of the objects from the scripts.
I would suggest to provide this object via `NashornEngine#put`, instead of via `StringBuilder`.

Reply:
As I said above: I don't really like the idea of injecting global objects. @qparis what do you think?

Reply:
I'm not sure what you mean? The access scope should be nearly the same as for your `environment` string approach, right?

Reply:
It's not about the scope. I just don't like that there would be an **object** which is "just there". The idea of environment variables as strings is quite common (e.g. in a shell). Therefore this approach somehow feels more "natural" for me.

Reply:
I understand where you're coming from, but I still think it's nicer in our case to directly inject the objects, instead of letting the user fetch them himself, especially because we already know them on the java side.

I would like to wait for @qparis to see what he thinks about this.